brave/brave-core
# Reland again android keychecker crash

Resolves https://github.com/brave/brave-browser/issues/28978


The difference with the previous PR is https://github.com/brave/brave-core/commit/9317cebe7a1ee2ae367a788beae33dbb87fef8c9  - I have removed the proguard rule.

The error which gave R8 wasn't clear enough:
```
Error: Discard checks failed.
The following items were not discarded
Item org.chromium.chrome.browser.preferences.ChromePreferenceKeyChecker was not discarded.
org.chromium.chrome.browser.preferences.ChromePreferenceKeyChecker
|- is referenced from:
|  org.chromium.chrome.browser.preferences.ChromePreferenceKeyChecker org.chromium.chrome.browser.preferences.BravePreferenceKeyChecker.getInstance()
|- is referenced in keep rule:
|  ../../brave/android/java/proguard.flags:7:1
​...
Caused by: [CIRCULAR REFERENCE: com.android.tools.r8.utils.AbortException: Discard checks failed.
```

We didn't see the issue on CI because it didn't happen on Nightly and Development channels, it happened only on Stable and Beta. 

I ensured it can be built and passes on-boarding on such configurations:
- Stable x86
- Stable x64
- Nightly x86
- Beta x86
- Stable arm32
- Nightly arm32
- Stable arm64.




## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Ensure Stable channel Release build is not broken:
`npm run build -- Release --target_arch=x86 --target_android_base=mono --target_os=android --target_android_output_format=apk`;
2. Ensure there is no crash on app start and possible to pass onboarding.
